src/lib/upload-forms.js

```diff
@@ -7,80 +6,69 @@ const log = require('./log');
 const insertOrReplace = require('./insert-or-replace');
 const pouch = require('./db');
 const warnUploadOverwrite = require('./warn-upload-overwrite');
+const {
+  getFormDir,
+  getFormFilePaths,
+  readTitleFrom,
+  readIdFrom
+} = require('./forms-utils');
 
 const SUPPORTED_PROPERTIES = ['context', 'icon', 'title', 'xml2sms', 'subject_key'];
 
-module.exports = (projectDir, subDirectory, options) => {
+module.exports = async (projectDir, subDirectory, options) => {
   const db = pouch();
   if (!options) options = {};
-
-  const formsDir = `${projectDir}/forms/${subDirectory}`;
+  const formsDir = getFormDir(projectDir, subDirectory);
   if(!fs.exists(formsDir)) {
     log.warn(`Forms dir not found: ${formsDir}`);
-    return Promise.resolve();
+    return;
```

Comment:
The return is safe because I changed the definition of the function to `async`, and async functions always return promises, either because an explicit promise is returned or because the JS engine wraps up any result in a promise, eg from the Node CLI:

```js
> (async function() { return; })().then(r=> console.log('Success! ' + r))
Promise { <pending> }
> Success! undefined

> (async function() { return Promise.resolve() })().then(r=> console.log('Success! ' + r))
Promise { <pending> }
> Success! undefined
```

Reply:
TIL! Thanks.